File: src/Programmers/코딩테스트 실전 대비 모의고사/1차/1.py
#!/usr/bin/python3
from itertools import repeat


def solution(X: str, Y: str):
    # range 9 to 0
    answer = ""
    for i in range(9, -1, -1):
        num = min(X.count(str(i)), Y.count(str(i)))
        answer += "".join(repeat(str(i), num))

    if answer == "":
        answer = "-1"
    if answer[0] == "0" and answer[1] == "0":
        answer = "0"
    return answer


print(solution("100", "2345"))  # -1
print(solution("100", "203045"))  # 0
print(solution("100", "123450"))  # 10

File: src/Programmers/코딩테스트 실전 대비 모의고사/2차/1.ts
/**
 * 학생 번호 sum == 0: 삼총사
 * 삼총사 만들 수 있는 방법 수
 */

function solution(numbers: number[]): number {
  var answer = 0;

  for (let a = 0; a < numbers.length; a++)
    for (let b = a + 1; b < numbers.length; b++)
      for (let c = b + 1; c < numbers.length; c++)
        if (numbers[a] + numbers[b] + numbers[c] == 0) answer += 1;
  return answer;
}

console.log(solution([-2, 3, 0, 2, -5])); // 2
console.log(solution([-3, -2, -1, 0, 1, 2, 3])); // 5
console.log(solution([-1, 1, -1, 1])); // 0
